sql-plugin/src/main/java/com/nvidia/spark/rapids/iceberg/spark/source/GpuMultiFileBatchReader.java

```diff
@@ -388,8 +399,10 @@ class ParquetCoalescingBatchReader extends MultiFileBatchReaderBase {
     protected FilePartitionReaderBase createRapidsReader(PartitionedFile[] pFiles,
         StructType partitionSchema) {
       ArrayList<ParquetSingleDataBlockMeta> clippedBlocks = new ArrayList<>();
-      files.values().forEach(fst -> {
-        FilteredParquetFileInfo filteredInfo = filterParquetBlocks(fst);
+      Arrays.stream(pFiles).forEach(pFile -> {
+        String partFilePathString = pFile.filePath().toString();
+        FileScanTask fst = files.get(partFilePathString);
+        FilteredParquetFileInfo filteredInfo = filterParquetBlocks(fst, partFilePathString);
```

Comment:
nit: Let's add some comments why we need to pass in ```partFilePathString``` other than using ```fst.file().path().toString()``` to get it.

https://github.com/NVIDIA/spark-rapids/issues/9697#issuecomment-1809757132

Reply:
We can use `toEncodedPathString(fst)` instead of this `partFilePathString ` here, but I don't want to do this encoding again since we already have the encoded path string in the `PartitionedFile`.